### DISCUSSION TITLE
Question: How to exclude a label from /issues rest api call?

### DISCUSSION THREAD

**Discussion by nick-shkaruba** [community/community#59520] [Programming Help] [answered]
### Body

Hello!

I want to list all issues in a repository without a label.

In https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#list-repository-issues I can see the `labels` parameter

I'm making a request with curl like so:
```
curl -s \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <token>"\
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/<owner>/<repo>/issues?labels=:rocket:+feature,:book:+documentation
```

Which gives me all the issues with feature and documentation labels.

But I actually want to exclude those labels. How do I do it?

My repository is private, so search rest api doesn't work for some reason

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Yarden-zamir** [comment, marked as answer]
According to their docs, it is not possible. A workaround you can use is grabbing all the issues and filtering the ones that don't have those labels. Here's an example of doing that using jq and gh cli instead of curl (should work the same with curl)

```bash
gh issue list --json labels,number \
	| jq '.[] | select(.labels | map(.name) | contains(["test", "registered_data"]) | not) | .number'
```
This gives me a list of issue numbers

**Reply by nick-shkaruba** [reply]
Good, thank you for the response. 
The one problem I have with your snippet is that the max amount of issues we can get from one api call is 100 according to the [docs](https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#list-repository-issues). But I have much more than that. I need a scalable solution.

I guess I can select issues with chunks, until I have something though! :)

**Reply by Yarden-zamir** [reply]
I have good news for you then, the github cli does manage to fetch more than the max from the api, I assume it handles the poaging internally
![Screen Shot 2023-06-29 at 14 43 05](https://github.com/community/community/assets/8178413/89a8bce5-fedd-4014-ac17-105a0b7a6942)
